examples/step-42/step-42.cc

```diff
@@ -876,8 +876,12 @@ namespace Step42
     output_dir = prm.get("output directory");
     if (output_dir != "" && *(output_dir.rbegin()) != '/')
       output_dir += "/";
-    const int ierr = mkdir(output_dir.c_str(), 0777);
-    AssertThrow(ierr == 0, ExcIO());
+
+    if (Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+      {
+        const int ierr = mkdir(output_dir.c_str(), 0777);
+        AssertThrow(ierr == 0, ExcIO());
```

Comment:
Do we want to allow that the directory already exists?

Reply:
Good point. I added a check for this.